packages/block-editor/src/hooks/utils.js

```diff
@@ -568,7 +568,7 @@ export function createBlockListBlockFilter( features ) {
 						useBlockProps,
 					} = feature;
 
-					const neededProps = {};
+					const neededProps = { clientId: props.clientId };
```

Comment:
I think this means the optimization on line 581 has no effect (`Object.keys( neededProps ).length` will always be greater than 1), which probably causes a degradation in performance in some other hooks, though I'm not sure how much of an issue it is.

Perhaps it could be set as a prop directly on the component rather than via `neededProps`. (@ellatrix might know this code better and be able to suggest a better alternative)

Reply:
Good catch!  It makes sense to move it to the component below the same way `name` is passed through.

Reply:
I've quickly made that change in https://github.com/WordPress/gutenberg/pull/57908/commits/df4a51858d2821b41fe0702fd309321679a497b7 in case others are reviewing the PR.

A quick smoke test didn't show any ill effects. I'll give it more of a test tomorrow morning.